hare/eligibility/oracle.go

```diff
@@ -446,28 +491,28 @@ func (o *Oracle) computeActiveSet(ctx context.Context, targetEpoch types.EpochID
 func (o *Oracle) computeActiveWeights(
 	targetEpoch types.EpochID,
 	activeSet []types.ATXID,
-) (map[types.NodeID]uint64, error) {
-	weightedActiveSet := make(map[types.NodeID]uint64)
+) (map[types.NodeID]identityWeight, error) {
+	identities := map[types.NodeID]identityWeight{}
 	for _, id := range activeSet {
-		atx, err := o.cdb.GetAtxHeader(id)
-		if err != nil {
-			return nil, fmt.Errorf("hare actives get ATX %s, epoch %d: %w", id, targetEpoch, err)
+		atx := o.atxsdata.Get(targetEpoch, id)
+		if atx == nil {
+			return nil, fmt.Errorf("oracle: missing atx in atxsdata %s/%s", targetEpoch, id.ShortString())
 		}
-		weightedActiveSet[atx.NodeID] = atx.GetWeight()
+		identities[atx.Node] = identityWeight{atx: id, weight: atx.Weight}
 	}
-	return weightedActiveSet, nil
+	return identities, nil
 }
 
 func (o *Oracle) activeSetFromRefBallots(epoch types.EpochID) ([]types.ATXID, error) {
 	beacon, err := o.beacons.GetBeacon(epoch)
 	if err != nil {
 		return nil, fmt.Errorf("get beacon: %w", err)
 	}
-	ballotsrst, err := ballots.AllFirstInEpoch(o.cdb, epoch)
+	ballotsrst, err := ballots.AllFirstInEpoch(o.db, epoch)
 	if err != nil {
 		return nil, fmt.Errorf("first in epoch %d: %w", epoch, err)
 	}
-	activeMap := make(map[types.ATXID]struct{})
+	activeMap := map[types.ATXID]struct{}{}
```

Comment:
Here it's more difficult to allocate a map that's big enough to not require reallocations, but we can assume that there is at least one unique ATX for every ballot I think?

```suggestion
	activeMap := make(map[types.ATXID]struct{}, len(ballotrst))
```

Reply:
yes we can assume. 

i am actually not sure how map is using size hint during initialization, will check it out